docs/csharp/whats-new/csharp-7-1.md

```diff
@@ -0,0 +1,176 @@
+
+Valid settings for "langVersion" element are:
+
+* ISO-1
+* ISO-2
+* 3
+* 4
+* 5
+* 6
+* 7
+* 7.1
+* default
+* latest
+
+The special strings "default" and "latest" resolve to the latest major
+and minor language versions installed on the build machine, respectively.
+
+This setting decouples installing new versions of the SDK and tools
+in your development environment from choosing to incorporate new language
+features in a project. You can install the latest SDK and tools on your
+build machine. Each project can be configured to use a specific version
+of the language for its build.
+
+## Async main
+
+An *async main* method enables you to use `await` in your `Main` method.
+Where previously you would need to write:
+
+```csharp
+static int Main()
+{
+    return DoAsyncWork().GetAwaiter().GetResult();
+}
+```
+
+You can know write:
+
+```csharp
+static async Task<int> Main()
+{
+    // This could also be replaced with the body
+    // DoAsyncWork, including its await expressions:
+    return await DoAsyncWork();
+}
+```
+
+If your program does not return an exit code, you can declare `Main` method
+that returns a `Task`:
+
+```csharp
+static async Task Main()
+{
+    await SomeAsyncMethod();
+}
+```
+
+You can read more about the details in the
+[async main](../programming-guid/main-and-command-args/index.md) topic
+in the programming guide.
+
+## Default value expressions
+
+When you initialize an object using a default value expression, you can
+omit the type from the default expression. Where you previously needed to
+type:
+
+```csharp
+Func<string, bool> whereClause = default(Func<string, bool>);
+```
+
+You can now write:
+
+```csharp
+Func<string, bool> whereClause = default;
+```
+
+There are many language constructs where default value expressions can make use
+of this new syntax. You can learn more in the topic on default value expressions
+in the topics on the [`default` keyword](../language-reference/keywords/default.md)
+and the topic on [default value expressions](../programming-guide/statements-expressions-operators/default-value-expressions.md).
+
+## Inferred tuple element names
+
+This feature is a small enhancement to the tuples feature introduced in
+C# 7.0. Many times that you initialize a tuple, the variables used for the
+right side of the assignment are the same as the names you'd like for the
+tuple elements:
+
+```csharp
+int count = 5;
+string label = "Colors used in the map";
+var pair = (count: count, label: label);
+```
+
+The names of tuple elements can be inferred from the variables used to initialize
+the tuple in C# 7.1:
+
+```csharp
+int count = 5;
+string label = "Colors used in the map";
+var pair = (count, label); // element names are "count" and "label"
+```
+
+You can learn more about the details for this feature in the [Tuples](../tuples.md)) topic.
+
+## Reference assembly generation
+
+There are two new compiler options that generate *reference only assemblies*:
```

Comment:
There are two new compiler options that --> Two new compiler options
reference only --> reference-only